Test failure in CI build 7137

The following test appears to have failed:

[#7137](https://circleci.com/gh/cockroachdb/cockroach/7137):

```
tb6627f72  127.0.0.1:0  13:31:32.350564  5.128µs    ·meta descriptor lookup       kv/dist_sender.go:571
tb6627f72  127.0.0.1:0  13:31:32.350572  668.739µs  ·sending RPC                  kv/dist_sender.go:477
tb6627f72  127.0.0.1:0  13:31:32.350600  0          ··sending to 127.0.0.1:52531  rpc/send.go:171

testdata/subquery:1: SELECT (SELECT 1)
--- FAIL: TestLogic (4.82s)
	logic_test.go:392: testdata/subquery:1: expected success, but found sql/parser/eval.go:1044: eval: unexpected expression: *parser.Subquery
=== RUN   TestPrivilege
--- PASS: TestPrivilege (0.00s)
=== RUN   TestPrivilegeValidate
--- PASS: TestPrivilegeValidate (0.00s)
=== RUN   TestSystemPrivilegeValidate
--- PASS: TestSystemPrivilegeValidate (0.00s)
=== RUN   TestAllocateIDs
--- PASS: TestAllocateIDs (0.00s)
=== RUN   TestValidateTableDesc
--- PASS: TestValidateTableDesc (0.01s)
=== RUN   TestColumnTypeSQLString
--- PASS: TestColumnTypeSQLString (0.00s)
FAIL
FAIL	github.com/cockroachdb/cockroach/sql	5.267s
=== RUN   TestDatumString
--- PASS: TestDatumString (0.00s)
=== RUN   TestPlaceholders
I0909 13:31:30.606894 857 base/context.go:155  setting up TLS from certificates directory: test_certs
I0909 13:31:30.608070 857 base/context.go:117  setting up TLS from certificates directory: test_certs
I0909 13:31:30.610750 857 rpc/clock_offset.go:155  monitoring cluster offset
I0909 13:31:30.611050 857 multiraft/multiraft.go:447  node 100000001 starting
I0909 13:31:30.611279 857 raft/raft.go:406  group 1 100000001 became follower at term 5
I0909 13:31:30.611382 857 raft/raft.go:219  group 1 newRaft 100000001 [peers: [100000001], term: 5, commit: 10, applied: 10, lastindex: 10, lastterm: 5]
I0909 13:31:30.611517 857 raft/raft.go:485  group 1 100000001 is starting a new election at term 5
--
=== RUN   TestNormalizeTableName
--- PASS: TestNormalizeTableName (0.00s)
=== RUN   TestNormalizeColumnName
--- PASS: TestNormalizeColumnName (0.00s)
=== RUN   TestNormalizeExpr
--- FAIL: TestNormalizeExpr (0.01s)
	normalize_test.go:90: (SELECT 1): sql/parser/eval.go:1044: eval: unexpected expression: *parser.Subquery
=== RUN   TestNormalizeExprError
--- PASS: TestNormalizeExprError (0.00s)
=== RUN   TestParse
--- PASS: TestParse (0.03s)
=== RUN   TestParse2
--- PASS: TestParse2 (0.01s)
=== RUN   TestParseSyntax
--- PASS: TestParseSyntax (0.00s)
=== RUN   TestParseError
--
--- PASS: TestFillArgs (0.01s)
=== RUN   TestFillArgsError
--- PASS: TestFillArgsError (0.00s)
=== RUN   TestWalkStmt
--- PASS: TestWalkStmt (0.00s)
FAIL
FAIL	github.com/cockroachdb/cockroach/sql/parser	0.102s
=== RUN   TestPrivilegeDecode
--- PASS: TestPrivilegeDecode (0.00s)
PASS
ok  	github.com/cockroachdb/cockroach/sql/privilege	0.012s
=== RUN   TestUpdateRangeAddressing
I0909 13:31:53.150960 1005 multiraft/multiraft.go:447  node 100000001 starting
I0909 13:31:53.151183 1005 storage/replica.go:1191  gossiping cluster id  from store 1, range 1
I0909 13:31:53.151465 1005 raft/raft.go:406  group 1 100000001 became follower at term 5
I0909 13:31:53.151564 1005 raft/raft.go:219  group 1 newRaft 100000001 [peers: [100000001], term: 5, commit: 10, applied: 10, lastindex: 10, lastterm: 5]
I0909 13:31:53.151794 1005 raft/raft.go:485  group 1 100000001 is starting a new election at term 5

```
Please assign, take a look and update the issue accordingly.

--- a/sql/parser/normalize.go
+++ b/sql/parser/normalize.go
@@ -359,6 +359,9 @@ func (v *isConstVisitor) Visit(expr Expr, pre bool) (Visitor, Expr) {
 		}
 
 		switch t := expr.(type) {
+		case *Subquery:
+			v.isConst = false
+			return nil, expr
 		case *FuncExpr:
 			// typeCheckFuncExpr populates t.fn.impure.
 			if _, err := typeCheckFuncExpr(t); err != nil || t.fn.impure {

